File: Stock_Maintenance_System_Ui/Stock_Maintenance_System_Ui/src/app/models/Product/UpdateProductRequest.ts
export interface UpdateProductRequest {
  productId: number;
  productName: string;
  companyId: number;
  categoryId: number;
  productCategoryId?: number;
  description?: string;
  mrp: number;
  salesPrice: number;
  totalQuantity: number;
  isActive: boolean;
  taxType?: string;       // default "GST"
  barCode?: string;       // default null
  brandName?: string;     // default null
  taxPercent?: number;           // default 18
}

File: Stock_Maintenance_System_Api/ApiRequest/UpdateProductRequest.cs
namespace InventorySystem_Api.ApiRequest;
public record UpdateProductRequest(
   int ProductId,
   string ProductName,
   int CompanyId,
   int CategoryId,
   int? ProductCategoryId,
   string? Description,
   decimal Mrp,
   decimal SalesPrice,
   int TotalQuantity,
   bool IsActive,
   decimal LandingPrice,
   string? SerialNo
);